core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerDataReader.java

```diff
@@ -83,10 +84,13 @@ public long pos() {
   }
 
   @Override
-  public void close() {
+  public void close() throws IOException {
     if (mClosed) {
       return;
     }
+    if (mReader != null) {
+      mReader.close();
+    }
```

Comment:
isn't this enough to ensure `mReader` is closed?
why do we need to move `mBlockWorker.createBlockReader()` to constructor of `BlockWorkerDataReader`?